core/ArchiveProcessor/Rules.php

```diff
@@ -356,4 +364,24 @@ public static function shouldProcessSegmentsWhenReArchivingReports()
     {
         return Config::getInstance()->General['rearchive_reports_in_past_exclude_segments'] != 1;
     }
+
+    public static function isSegmentPluginArchivingDisabled($pluginName, $siteId = null)
+    {
+
+        $pluginArchivingSetting = GeneralConfig::getConfigValue('disable_archiving_segment_for_plugins', $siteId);
+
+        if (empty($pluginArchivingSetting)) {
+            return false;
+        }
+
+        if (is_string($pluginArchivingSetting)) {
+            $pluginArchivingSetting = explode(",", "$pluginArchivingSetting");
+        }
+
+        if (in_array($pluginName, $pluginArchivingSetting)) {
+            return true;
+        }
+        return false;
```

Comment:
This can be simplified as:

```suggestion
        return in_array($pluginName, $pluginArchivingSetting);
```